### DISCUSSION TITLE
when debugging, how can I read stdout and stderr of the program and send somthing to stdin?

### DISCUSSION THREAD

**Discussion by meg4152** [rizinorg/cutter#2951] [Q&A]
When debugging, how can I read stdout and stderr of the program and send somthing to stdin?
for example, when the program runs `read(0,buffer,size)` , then the program just stops and I can't find any way to send the input.
I can't also get the stdout of the program.

**Comment by ret2libc** [comment]
Could you give more precise instructions on which commands you do, the binary maybe, etc. ?

**Comment by ITAYC0HEN** [comment]
In the console widget, we have a Debuggee Input mode:
![image](https://user-images.githubusercontent.com/20182642/171115581-16707708-eb71-4139-bfd0-bf3c182dbe41.png)

**Comment by meg4152** [comment]
Thank you very much for answering me! And I find it now, as shown in the picture below.
![Capture d’écran du 2022-06-05 14-17-03](https://user-images.githubusercontent.com/40293541/172038284-fc6ab879-b141-4d40-9178-a1189d7ff9a9.png)
 
But I also find a problem: the glibc won't output the stdout buffer immediately: the string "please input : " isn't printed before asking for input.
![Capture d’écran du 2022-06-05 14-20-09](https://user-images.githubusercontent.com/40293541/172038384-43a0bd5a-c224-4b42-b980-1994f721c834.png)
 
And I wonder if there is a way to input some bytes not printable. As in the picture I tried to input "\x30\x31\n" (i hope for a way to input 0x30, 0x31, 0xa to the program).

**Reply by ITAYC0HEN** [reply]
for this you'll have to use an rz-run script 
(see man rz-run, or https://www.mankier.com/1/rz-run, or https://book.rizin.re/tools/rz-run/intro.html)
